charts/openshift-metering/templates/openshift-reporting/report-queries/pod-memory.yaml

```diff
@@ -362,22 +362,21 @@ spec:
     ), namespace_memory_request AS (
       {| renderReportQuery .Report.Inputs.NamespaceMemoryRequestQueryName . |}
     )
-    SELECT
-      timestamp '{| default .Report.ReportingStart .Report.Inputs.ReportingStart| prestoTimestamp |}' AS period_start,
-      timestamp '{| default .Report.ReportingEnd .Report.Inputs.ReportingEnd | prestoTimestamp |}' AS period_end,
-      usage.namespace,
-      usage.pod_usage_memory_byte_seconds,
-      request.pod_request_memory_byte_seconds,
-      usage.pod_usage_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_usage_percent,
-      request.pod_request_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_request_percent,
-      capacity.total_cluster_capacity_memory_byte_seconds
-    FROM namespace_memory_request as request
-    JOIN namespace_memory_usage as usage
-      ON request.period_start = usage.period_start
-      AND request.period_end = usage.period_end
-      AND request.namespace = usage.namespace
-    JOIN cluster_memory_capacity as capacity
-      ON capacity.period_start = request.period_start
-      AND capacity.period_end = request.period_end
-      AND capacity.period_start = usage.period_start
-      AND capacity.period_end = usage.period_end
+  SELECT
+    timestamp '2020-08-31 16:11:00.000' AS period_start,
+    timestamp '2020-08-31 16:52:00.000' AS period_end,
+    usage.namespace AS usage_namespace,
+    request.namespace AS request_namespace,
+    usage.pod_usage_memory_byte_seconds,
+    request.pod_request_memory_byte_seconds,
+    usage.pod_usage_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_usage_percent,
+    request.pod_request_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_request_percent,
+    capacity.total_cluster_capacity_memory_byte_seconds
+  FROM namespace_memory_usage as usage
+  JOIN cluster_memory_capacity as capacity
+    ON capacity.period_start = usage.period_start
+    AND capacity.period_end = usage.period_end
+  FULL OUTER JOIN namespace_memory_request as request
```

Comment:
I feel like this should be a left join and not an outer join but I'm open to other opinions. I think it's a safe assumption that we're always going to have more or equal row results in the usage table vs. the memory table. I can't think of a scenario where Pods in a namespace that declare a memory request but don't actually utilize any memory would need to be accounted for.

Reply:
Outer is needed to bring back NULL. Report looks good this way. If it's not broke ....

Reply:
I'm not sure how you can make the claim that left will not work here when we've already confirmed that it does locally.

Pulling down your change to first join the cluster_memory_capacity table locally, and substituting the `FULL OUTER JOIN` with a `LEFT JOIN` results in the following rendered ReportQuery:
```
WITH cluster_cpu_capacity AS (
  SELECT
  timestamp '2020-08-31 16:11:00.000' AS period_start,
  timestamp '2020-08-31 21:46:00.000' AS period_end,
  sum(cpu_core_seconds)  as total_cluster_capacity_cpu_core_seconds,
  avg(cpu_cores) as avg_cluster_capacity_cpu_cores,
  avg(node_count) AS avg_node_count
  FROM hive.metering.datasource_emily_cluster_cpu_capacity_raw
  WHERE "timestamp"  >= timestamp '2020-08-31 16:11:00.000'
  AND "timestamp" < timestamp '2020-08-31 21:46:00.000'
  AND dt >= '2020-08-31'
  AND dt <= '2020-08-31'

), namespace_cpu_usage AS (
  SELECT
  timestamp '2020-08-31 16:11:00.000' AS period_start,
  timestamp '2020-08-31 21:46:00.000' AS period_end,
  namespace,
  sum(pod_usage_cpu_core_seconds) as pod_usage_cpu_core_seconds
FROM hive.metering.datasource_emily_pod_cpu_usage_raw
WHERE "timestamp" >= timestamp '2020-08-31 16:11:00.000'
AND "timestamp" < timestamp '2020-08-31 21:46:00.000'
AND dt >= '2020-08-31'
AND dt <= '2020-08-31'
GROUP BY namespace

), namespace_cpu_request AS (
  SELECT
  timestamp '2020-08-31 16:11:00.000' AS period_start,
  timestamp '2020-08-31 21:46:00.000' AS period_end,
  namespace,
  sum(pod_request_cpu_core_seconds) as pod_request_cpu_core_seconds
FROM hive.metering.datasource_emily_pod_cpu_request_raw
WHERE "timestamp" >= timestamp '2020-08-31 16:11:00.000'
AND "timestamp" < timestamp '2020-08-31 21:46:00.000'
AND dt >= '2020-08-31'
AND dt <= '2020-08-31'
GROUP BY namespace

)
SELECT
  timestamp '2020-08-31 16:11:00.000' AS period_start,
  timestamp '2020-08-31 21:46:00.000' AS period_end,
  usage.namespace,
  usage.pod_usage_cpu_core_seconds,
  request.pod_request_cpu_core_seconds,
  usage.pod_usage_cpu_core_seconds / capacity.total_cluster_capacity_cpu_core_seconds AS pod_cpu_usage_percent,
  request.pod_request_cpu_core_seconds / capacity.total_cluster_capacity_cpu_core_seconds AS pod_cpu_request_percent,
  capacity.total_cluster_capacity_cpu_core_seconds
FROM namespace_cpu_usage as usage
JOIN cluster_cpu_capacity as capacity
  ON capacity.period_start = usage.period_start
  AND capacity.period_end = usage.period_end
LEFT JOIN namespace_cpu_request as request
  ON request.period_start = usage.period_start
  AND request.period_end = usage.period_end
  AND request.namespace = usage.namespace
```

When I run that rendered query against a local Presto instance, I get the following results back:
```
      period_start       |       period_end        |                    namespace                     | pod_usage_cpu_core_seconds | pod_request_cpu_core_seconds | pod_cpu_usage_percent | pod_cpu_request_percent | total_cluster_capacity_cpu_core_seconds 
-------------------------+-------------------------+--------------------------------------------------+----------------------------+------------------------------+-----------------------+-------------------------+-----------------------------------------
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-controller-manager-operator            |                  107.30322 |            200.9999999999998 | 1.5742843309859153E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-controller-manager                     |          97.32299999999996 |                       6030.0 | 1.4278609154929573E-4 |    0.008846830985915493 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-controller-manager-operator       |         402.31757999999996 |            200.9999999999998 |  5.902546654929577E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-insights                               |         27.035279999999997 |            200.9999999999998 | 3.9664436619718304E-5 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cloud-credential-operator              |                   73.44162 |                        603.0 | 1.0774885563380282E-4 |    8.846830985915492E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-controller-manager                |          682.7161199999999 |                      15075.0 | 0.0010016374999999998 |     0.02211707746478873 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-version                        |                  138.34602 |            401.9999999999996 |  2.029724471830986E-4 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-storage-operator               |         114.93011999999999 |                        603.0 | 1.6861813380281688E-4 |    8.846830985915492E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-machine-api                            |         192.46920000000003 |                       2412.0 |  2.823785211267606E-4 |    0.003538732394366197 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-csi-drivers                    |         104.84778000000001 |            6317.999999999988 | 1.5382596830985917E-4 |    0.009269366197183082 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-scheduler-operator                |          394.2830400000001 |                         NULL |  5.784669014084508E-4 |                    NULL |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-node-tuning-operator           |                   68.45316 |           1905.0000000000036 | 1.0043010563380281E-4 |   0.0027948943661971885 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-dns                                    |         443.25606000000005 |           11076.000000000002 |  6.503169894366198E-4 |    0.016250000000000004 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-authentication                         |                    99.8397 |            401.9999999999996 | 1.4647843309859154E-4 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-console-operator                       |                  103.51488 |            200.9999999999998 | 1.5187042253521127E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-marketplace                            |          644.7250799999998 |           2366.4000000000037 |  9.458994718309857E-4 |   0.0034718309859154985 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-samples-operator               |                    33.3009 |            401.9999999999996 |  4.885695422535211E-5 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-sdn                                    |         1372.3166399999998 |           35781.600000000006 |  0.002013375352112676 |    0.052496478873239445 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | tflannag                                         |          5453.963519999998 |                      51585.0 |  0.008001707042253519 |     0.07568221830985916 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-apiserver                              |          3301.081559999999 |           6633.0000000000055 |  0.004843136091549295 |     0.00973151408450705 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-console                                |          52.86078000000001 |            803.9999999999992 |  7.755396126760566E-5 |   0.0011795774647887312 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-etcd-operator                          |                  329.10648 |            200.9999999999998 |   4.82844014084507E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-network-operator                       |          69.86903999999998 |            200.9999999999998 | 1.0250739436619716E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-storage-version-migrator-operator |         161.05716000000004 |            200.9999999999998 |  2.362927816901409E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-cluster-machine-approver               |         12.775020000000005 |            401.9999999999996 | 1.8742693661971837E-5 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-apiserver-operator                |          529.9073400000001 |            200.9999999999998 |  7.774462147887325E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-oauth-apiserver                        |                  356.48652 |                       9045.0 |  5.230142605633803E-4 |    0.013270246478873239 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-apiserver-operator                     |         289.28837999999996 |            200.9999999999998 |   4.24425440140845E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-authentication-operator                |          648.8123999999999 |            200.9999999999998 |  9.518961267605632E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-etcd                                   |         15301.390320000004 |           35577.000000000015 |   0.02244922288732395 |     0.05219630281690143 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-dns-operator                           |         15.287579999999998 |            401.9999999999996 |  2.242896126760563E-5 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-service-ca-operator                    |                    72.3528 |            200.9999999999998 | 1.0615140845070423E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | btofel-reportexpiration                          |          9743.703600000003 |                      67011.0 |  0.014295339788732398 |     0.09831426056338029 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-ingress-operator                       |         20.292060000000003 |            401.9999999999996 |   2.97712147887324E-5 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-image-registry                         |         147.74423999999996 |            5924.999999999989 | 2.1676091549295768E-4 |    0.008692781690140828 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-apiserver                         |         15398.703300000001 |           31203.000000000025 |  0.022591994278169016 |    0.045779049295774685 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-scheduler                         |         206.53812000000008 |            10251.00000000001 | 3.0301954225352124E-4 |     0.01503961267605635 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-config-operator                        |         113.97425999999999 |            200.9999999999998 |  1.672157570422535E-4 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-operator-lifecycle-manager             |         1957.1840399999999 |            401.9999999999996 | 0.0028714554577464785 |    5.897887323943656E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | emily                                            |                 9494.59206 |                      67335.0 |  0.013929859242957747 |     0.09878961267605633 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-metering                               |          9005.888099999998 |                      36936.0 |   0.01321286399647887 |     0.05419014084507042 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-ingress                                |         119.37558000000003 |                       4020.0 |  1.751402288732395E-4 |    0.005897887323943662 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-multus                                 |         357.51401999999996 |            6318.000000000002 |  5.245217429577464E-4 |      0.0092693661971831 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-kube-storage-version-migrator          |                    2.46096 |                       2010.0 |   3.61056338028169E-6 |    0.002948943661971831 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-machine-config-operator                |                   722.2314 |            8826.000000000007 | 0.0010596117957746478 |    0.012948943661971841 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-service-ca                             |                    59.1051 |            200.9999999999998 |  8.671522887323944E-5 |    2.948943661971828E-4 |                                681600.0 
 2020-08-31 16:11:00.000 | 2020-08-31 21:46:00.000 | openshift-monitoring                             |          7355.115719999998 |            6116.520000000013 |  0.010790956161971827 |    0.008973767605633821 |                                681600.0 
(47 rows)
```

Looking at the above results, we can see that the openshift-cluster-samples-operator namespace appears in that queries results.